### ISSUE TITLE
Simplify Build System

### ISSUE THREAD

**Issue by tommy-waltmann** [glotzerlab/freud#924]
**Is your feature request related to a problem? Please describe.**

Since `scikit-build`'s 0.13.0 release, the flags which were hard-coded to not generate cython docstrings in CMake release builds are now gone. We had a workaround for this (#694, #843) where we created our own CMake build type and used that as the default build type. All that extra build logic can now be removed.

**Describe the solution you'd like**

We simply need to reqiure `scikit-build>=0.13.1` and update the CMake build scripts.

**Comment by bdice** [comment]
We'll also need to bump the oldest requirements and `pyproject.toml` beyond the updates in #920:
https://github.com/glotzerlab/freud/blob/30e2f333f67958e1625918b2f78986f46edfdb86/.circleci/ci-oldest-reqs.txt#L12
https://github.com/glotzerlab/freud/blob/30e2f333f67958e1625918b2f78986f46edfdb86/pyproject.toml#L2

**Comment by joaander** [comment]
What if a user installs using e.g. the ubuntu provided `scikit-build` package (0.11.1 in the most recent ubuntu)? https://packages.ubuntu.com/hirsute/python3-skbuild

Will freud still build and install, sans docstrings, or will it not build at all?

**Comment by bdice** [comment]
I think that `Release` builds work but do not include docstrings. I wrote in #694 that the build worked but docs were broken. I hid the v2.4.0 release from our ReadTheDocs version list because it was basically empty. Also our CI is currently building for Python 3.6 with scikit-build 0.10.0 (see pinning above).

**Comment by vyasr** [comment]
Perhaps I'm misunderstanding, but I think Josh is asking what will happen if we bump the required scikit-build version beyond the minimum available on Ubuntu, not whether freud installations work with a lower scikit-build version. I think the answer is that since we're using pyproject.toml it will simply pip install the minimum required scikit-build version unless you disable build isolation or pep517. If you pass either `--no-build-isolation` or `--no-use-pep517` my guess is that you'll just get a build without docs.

**Comment by joaander** [comment]
Let me rephrase the question. Will you be placing a hard `>=0.13.1` requirement such that freud will refuse to build on systems with a preinstalled old scikit-build.

**Comment by joaander** [comment]
> Perhaps I'm misunderstanding, but I think Josh is asking what will happen if we bump the required scikit-build version beyond the minimum available on Ubuntu, not whether freud installations work with a lower scikit-build version. I think the answer is that since we're using pyproject.toml it will simply pip install the minimum required scikit-build version unless you disable build isolation or pep517. If you pass either `--no-build-isolation` or `--no-use-pep517` it should just use the older version and you'll get a build without docs.

OK, that answers it. Builds that use build isolation won't work in this sort of config as they will be built against the latest numpy ABI which is most likely not compatible with the numpy provided by the system package manager.

**Comment by bdice** [comment]
@joaander We build against [oldest-supported-numpy](https://pypi.org/project/oldest-supported-numpy/) with build isolation for this reason:
https://github.com/glotzerlab/freud/blob/30e2f333f67958e1625918b2f78986f46edfdb86/pyproject.toml#L2

**Comment by vyasr** [comment]
`scikit-build` is a build requirement for us, but not an install requirement. If the builds that you are doing are using `--no-build-isolation` and `--no-use-pep517` I _think_ they will just ignore pyproject.toml and not try to install anything. If scikit-build is already installed via apt, installing should jump straight into setup.py and just work (it should fail if scikit-build was not already installed).

**Comment by joaander** [comment]
> @joaander We build against [oldest-supported-numpy](https://pypi.org/project/oldest-supported-numpy/) with build isolation for this reason:
> 
> https://github.com/glotzerlab/freud/blob/30e2f333f67958e1625918b2f78986f46edfdb86/pyproject.toml#L2

Right, I forgot about that.
